nomenclature/processor/region.py

```diff
@@ -349,7 +350,22 @@ def from_directory(cls, path: DirectoryPath, dsd: DataStructureDefinition):
         """
         mapping_dict: Dict[str, RegionAggregationMapping] = {}
         errors: List[ErrorWrapper] = []
-        for file in (f for f in path.glob("**/*") if f.suffix in {".yaml", ".yml"}):
+
+        mapping_files = [f for f in path.glob("**/*") if f.suffix in {".yaml", ".yml"}]
+
+        if (file := path.parent / ".nomenclature.yaml").exists():
```

Comment:
Not ideal if the nomenclature-config file is parsed several times - but does it mean that the cloned repo is also pulled several times?

Reply:
It's not ideal that's true. One way around it would be to separately create a `NomenclatureConfig` object once and pass it to `DataStructureDefinition` and `RegionProcessor`. 
I could implement that. It's just something that would need to be added to every workflow though ...

Reply:
True, good idea, that should work.